.github/workflows/deploy.yml

```diff
@@ -0,0 +1,58 @@
+name: Deployment
+
+on:
+  # https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch
+  workflow_dispatch:
+    branch:
+      description: 'The branch to deploy'
+      default: 'main'
+      required: false
+      type: string
+
+env:
+  NODE_ENV: development
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  NETLIFY_TOKEN: ${{ secrets.NETLIFY_TOKEN }}
+  NETLIFY_SITE_ID: ${{ secrets.NETLIFY_SITE_ID }}
+  FIGMA_TOKEN: ${{ secrets.FIGMA_TOKEN }}
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    environment: production
+    steps:
+      - uses: actions/checkout@v3
+        ref: ${{ github.event.inputs.branch }}
+        # https://github.com/actions/checkout/blob/main/src/input-helper.ts#L103
+        submodules: 'TRUE'
+
+      - name: Setup
+        uses: actions/setup-node@v3
+        with:
+          node-version: '14'
+          cache: 'yarn'
+
+      # https://github.com/actions/setup-node/issues/182
+      - name: Install Yarn
+        run: npm install --global yarn
+
+      - name: Install
+        run: yarn
+
+      - name: Test
+        run: yarn test
```

Comment:
Why is `yarn test` ever needed on a manual deployment? The `main` branch always tests with "success" status.

Reply:
This originally assumed that `main` branch is not the only target..